src/context/toast.test.tsx

```diff
@@ -0,0 +1,228 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { render, fireEvent, waitFor, act } from "test_utils/test-utils";
+import { RequireAtMostOne } from "types/utils";
+import { RenderFakeToastContext } from "./__tests__/toast";
+import { useToastContext, ToastProvider } from "./toast";
+
+type useToastContextProps = ReturnType<typeof useToastContext>;
+
+// This type requires that the component only accepts one of the props
+type UseToastComponentProps = RequireAtMostOne<
+  {
+    [K in keyof useToastContextProps]: Parameters<useToastContextProps[K]>;
+  }
+>;
+
+describe("Real Toast", () => {
+  // Since useToastContext relies on the toastProvider which in turn relies on the react.createPortal api we cannot test it directly
+  // because the react.createPortal api is not available in the testing-library/react-hooks test environment. So we need to create a wrapper
+  // component that will render the toastProvider and useToastContext and test the useToastContext hook internally.
+  const UseToastComponent: React.FC<UseToastComponentProps> = (props) => {
```

Comment:
This is a test fixture around the useToastContext hook that mimics the implementation of a component using this hook.

Reply:
nit: this is kind of a confusing component name since we usually start our hooks with `use`